mllib/src/main/scala/org/apache/spark/ml/feature/IDF.scala

```diff
@@ -178,10 +188,19 @@ object IDFModel extends MLReadable[IDFModel] {
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
       val dataPath = new Path(path, "data").toString
       val data = sparkSession.read.parquet(dataPath)
-      val Row(idf: Vector) = MLUtils.convertVectorColumnsToML(data, "idf")
-        .select("idf")
-        .head()
-      val model = new IDFModel(metadata.uid, new feature.IDFModel(OldVectors.fromML(idf)))
+
+      val model = if (majorVersion(metadata.sparkVersion) >= 3) {
+        val Row(idf: Vector, df: Seq[_], numDocs: Long) = data.select("idf", "docFreq", "numDocs")
+          .head()
+        new IDFModel(metadata.uid, new feature.IDFModel(OldVectors.fromML(idf),
+          df.asInstanceOf[Seq[Long]].toArray, numDocs))
+      } else {
+        val Row(idf: Vector) = MLUtils.convertVectorColumnsToML(data, "idf")
+          .select("idf")
+          .head()
+        new IDFModel(metadata.uid,
```

Comment:
Yeah I guess in this case there is no way to know what the original doc frequencies were. Another option is to make these new fields an `Option` but I think it's not worth the indirection. It will be rare that someone loads an old model with new Spark version, and will be obvious that this data isn't present if they try to access it.

Reply:
Yes.

Another problem with returning option is that, people using the class from spark 3 onwards, will get an option. And it would be confusing because it may mean that docFreq may not be present. But it always would be. We would then need to have doc explaining the scenario of None,complicating the doc. 

Secondly it would need to remain consistent with other models. Where all new fields added should return option.